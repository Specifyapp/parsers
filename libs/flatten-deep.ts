export function flattenObject(obj: Record<string, any>) {
  const flattened: Record<string, any> = {};

  Object.keys(obj).forEach(key => {
    if (typeof obj[key] === 'object' && obj[key] !== null) {
      Object.assign(flattened, flattenObject(obj[key]));
    } else {
      flattened[key] = obj[key];
    }
  });

  return flattened;
}
