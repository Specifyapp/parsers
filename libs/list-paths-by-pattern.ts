import { has, get } from 'lodash';

export default function listPathsByPattern<T extends object>(
  base: T | Array<T>,
  pattern: string,
): Array<string> {
  try {
    if (!pattern.includes('[*]')) return has(base, pattern) ? [pattern] : [];
    return pattern
      .split(/(?=\[\*])|(?<=\[\*])/g) // split without remove the delimiter
      .reduce<Array<string>>(
        (paths, item) => {
          if (item === '[*]') {
            return paths
              .map(basePath => {
                const iteration = basePath ? get(base, basePath) : base;
                if (!Array.isArray(iteration)) return [];
                return Array(iteration.length)
                  .fill(0)
                  .reduce(
                    (acc, _, index) =>
                      has(base, `${basePath}[${index}]`) ? [...acc, `${basePath}[${index}]`] : acc,
                    [],
                  );
              })
              .flat(2);
          } else {
            return paths.reduce<Array<string>>(
              (acc, elm) => (has(base, `${elm}${item}`) ? [...acc, `${elm}${item}`] : acc),
              [],
            );
          }
        },
        [''],
      );
  } catch (err) {
    throw err;
  }
}
