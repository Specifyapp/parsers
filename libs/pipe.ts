type UnknownFunction = (...params: unknown[]) => unknown;

// Copied from https://github.com/gcanti/fp-ts/blob/1.15.0/src/function.ts#L209 with some
// modifications to make the first function allow multiple parameters.
export function pipe<A extends unknown[], B>(ab: (this: void, ...a: A) => B): (...args: A) => B;
export function pipe<A extends unknown[], B, C>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
): (...args: A) => C;
export function pipe<A extends unknown[], B, C, D>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
): (...args: A) => D;
export function pipe<A extends unknown[], B, C, D, E>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
): (...args: A) => E;
export function pipe<A extends unknown[], B, C, D, E, F>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
  ef: (this: void, e: E) => F,
): (...args: A) => F;
export function pipe<A extends unknown[], B, C, D, E, F, G>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
  ef: (this: void, e: E) => F,
  fg: (this: void, f: F) => G,
): (...args: A) => G;
export function pipe<A extends unknown[], B, C, D, E, F, G, H>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
  ef: (this: void, e: E) => F,
  fg: (this: void, f: F) => G,
  gh: (this: void, g: G) => H,
): (...args: A) => H;
export function pipe<A extends unknown[], B, C, D, E, F, G, H, I>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
  ef: (this: void, e: E) => F,
  fg: (this: void, f: F) => G,
  gh: (this: void, g: G) => H,
  hi: (this: void, h: H) => I,
): (...args: A) => I;
export function pipe<A extends unknown[], B, C, D, E, F, G, H, I, J>(
  ab: (this: void, ...a: A) => B,
  bc: (this: void, b: B) => C,
  cd: (this: void, c: C) => D,
  de: (this: void, d: D) => E,
  ef: (this: void, e: E) => F,
  fg: (this: void, f: F) => G,
  gh: (this: void, g: G) => H,
  hi: (this: void, h: H) => I,
  ij: (this: void, i: I) => J,
): (...args: A) => J;

export function pipe(...fns: UnknownFunction[]): UnknownFunction {
  return (...initialParams: unknown[]): unknown =>
    fns.reduce<unknown>((value, fn, index) => {
      const params = index === 0 ? (value as unknown[]) : [value];
      return fn(...params);
    }, initialParams);
}
