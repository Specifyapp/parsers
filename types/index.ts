export * from './tokens';
export * from './Providers';
export * from './utils/utils';
