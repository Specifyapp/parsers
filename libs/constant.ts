import { TokensType } from '../types';

export const tokenTypesList: Array<TokensType> = [
  'border',
  'color',
  'gradient',
  'duration',
  'measurement',
  'opacity',
  'shadow',
  'textStyle',
  'depth',
];
