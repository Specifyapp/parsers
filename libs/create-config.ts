import { Parser, Token } from '../types';

type UnknownParser<F> = { options: unknown; fn: F; name: string };
type ReturnPipe<T> = {
  run: (tokens: Array<Token>) => T;
  build: () => unknown;
};

export function createConfig<A extends unknown[], B>(
  ab: UnknownParser<(...a: A) => B>,
): ReturnPipe<B>;
export function createConfig<A extends unknown[], B, C>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
): ReturnPipe<C>;
export function createConfig<A extends unknown[], B, C, D>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
): ReturnPipe<D>;
export function createConfig<A extends unknown[], B, C, D, E>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
): ReturnPipe<E>;
export function createConfig<A extends unknown[], B, C, D, E, F>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
  ef: UnknownParser<(e: E) => F>,
): ReturnPipe<F>;
export function createConfig<A extends unknown[], B, C, D, E, F, G>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
  ef: UnknownParser<(e: E) => F>,
  fg: UnknownParser<(f: F) => G>,
): ReturnPipe<G>;
export function createConfig<A extends unknown[], B, C, D, E, F, G, H>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
  ef: UnknownParser<(e: E) => F>,
  fg: UnknownParser<(f: F) => G>,
  gh: UnknownParser<(g: G) => H>,
): ReturnPipe<H>;
export function createConfig<A extends unknown[], B, C, D, E, F, G, H, I>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
  ef: UnknownParser<(e: E) => F>,
  fg: UnknownParser<(f: F) => G>,
  gh: UnknownParser<(g: G) => H>,
  hi: UnknownParser<(h: H) => I>,
): ReturnPipe<I>;
export function createConfig<A extends unknown[], B, C, D, E, F, G, H, I, J>(
  ab: UnknownParser<(...a: A) => B>,
  bc: UnknownParser<(b: B) => C>,
  cd: UnknownParser<(c: C) => D>,
  de: UnknownParser<(d: D) => E>,
  ef: UnknownParser<(e: E) => F>,
  fg: UnknownParser<(f: F) => G>,
  gh: UnknownParser<(g: G) => H>,
  hi: UnknownParser<(h: H) => I>,
  ij: UnknownParser<(i: I) => J>,
): ReturnPipe<J>;

export function createConfig(...parsers: Array<UnknownParser<(...params: unknown[]) => unknown>>): {
  run: (tokens: Array<Token>) => unknown;
  build: () => unknown;
} {
  return {
    build: () =>
      parsers.map(parser => {
        const step: Parser = {
          name: parser.name,
        };
        if (parser.options) {
          step.options = parser.options;
        }
        return step;
      }),
    run: (tokens: Array<Token>): unknown => {
      return parsers.reduce<unknown>(async (value, parser) => parser.fn(await value), tokens);
    },
  };
}
