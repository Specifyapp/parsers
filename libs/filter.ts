import { AtLeast, TokensType, TokenTypeFromTokenTypeName } from '../types';

export function filter<T extends TokensType>(
  tokens: Array<AtLeast<{ type: TokensType }>>,
  type: T,
) {
  return tokens.filter(token => token.type === type) as Array<TokenTypeFromTokenTypeName<T>>;
}
