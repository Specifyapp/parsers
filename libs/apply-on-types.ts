import { tokenTypesList } from './constant';
import { TokensType } from '../types';

export function getTokenTypesToApplyFn<T extends { filter?: { types?: Array<TokensType> } }>(
  options: T,
) {
  if (!options?.filter?.types) {
    return tokenTypesList;
  }
  return tokenTypesList.filter(tokenType => options.filter!.types!.includes(tokenType));
}
