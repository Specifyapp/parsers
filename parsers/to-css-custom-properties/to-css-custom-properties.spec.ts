import libs from '../global-libs';
import * as seeds from '../../seeds.json';
import toCss, { OptionsType } from './to-css-custom-properties.parser';
import { ColorToken, ColorValue, Shadow, ShadowToken, Token, MeasurementToken } from '../../types';

describe('To css', () => {
  it('Get tokens - apply parsers', async done => {
    const result = await toCss(seeds.tokens as Array<Token>, undefined, libs);
    expect(typeof result).toEqual('string');
    const color = seeds.tokens.find((token: Token) => token.type === 'color') as Token;
    const measurement = seeds.tokens.find(
      (token: Token) => token.type === 'measurement',
    ) as MeasurementToken;
    const shadow = seeds.tokens.find((token: Token) => token.type === 'shadow') as ShadowToken;

    const shadowValue = shadow.value.reduce((acc: string, value: Shadow) => {
      const { color, offsetX, offsetY, blur, isInner } = value;
      const x = offsetX.value;
      const y = offsetY.value;
      const bl = blur.value;
      const { r, g, b, a } = color.value;
      const innerText = isInner ? 'inset ' : '';
      if (acc === '') {
        return `${innerText} ${x.measure}${x.unit} ${y.measure}${y.unit} ${bl.measure}${bl.unit} rgba(${r}, ${g}, ${b}, ${a})`;
      }

      return `${acc}, ${innerText}${x.measure}${x.unit} ${y.measure}${y.unit} ${bl.measure}${bl.unit} rgba(${r}, ${g}, ${b}, ${a})`;
    }, '');

    expect(
      result.includes(
        `${libs._.kebabCase(color.name)}: ${libs
          .tinycolor(color.value as ColorValue)
          .toString('rgb')}`,
      ),
    ).toBe(true);
    expect(result.includes(`${libs._.kebabCase(shadow.name)}:${shadowValue}`)).toBe(true);
    expect(
      result.includes(
        `${libs._.kebabCase(measurement.name)}: ${measurement.value.measure}${
          measurement.value.unit
        }`,
      ),
    ).toBe(true);
    done();
  });
  it('Get tokens - apply parsers - with options', async done => {
    const options: OptionsType = {
      formatName: 'snakeCase'!,
      formatTokens: { color: 'hsl' },
    };
    const result = await toCss(seeds.tokens as Array<Token>, options, libs);
    const color = seeds.tokens.find((token: Token) => token.type === 'color') as ColorToken;
    const measurement = seeds.tokens.find(
      (token: Token) => token.type === 'measurement',
    ) as MeasurementToken;

    const fnFormatColor = libs._[options.formatName!](color.name);
    expect(
      result.includes(
        `${fnFormatColor}: ${libs
          .tinycolor(color.value as ColorValue)
          .toString(options.formatTokens?.color)}`,
      ),
    ).toBe(true);

    const fnFormatMeasurement = libs._[options.formatName!](measurement.name);
    expect(
      result.includes(
        `${fnFormatMeasurement}: ${measurement.value.measure}${measurement.value.unit}`,
      ),
    ).toBe(true);
    done();
  });

  it('Get tokens - apply parsers - with custom selector', async done => {
    const options: OptionsType = {
      formatName: 'snakeCase'!,
      formatTokens: { color: 'hsl' },
      formatConfig: {
        selector: 'body[data-theme="light"]',
      },
    };
    const result = await toCss(seeds.tokens as Array<Token>, options, libs);
    const color = seeds.tokens.find((token: Token) => token.type === 'color') as ColorToken;
    const measurement = seeds.tokens.find(
      (token: Token) => token.type === 'measurement',
    ) as MeasurementToken;

    const fnFormatColor = libs._[options.formatName!](color.name);
    expect(
      result.includes(
        `${fnFormatColor}: ${libs
          .tinycolor(color.value as ColorValue)
          .toString(options.formatTokens?.color)}`,
      ),
    ).toBe(true);

    const fnFormatMeasurement = libs._[options.formatName!](measurement.name);
    expect(
      result.includes(
        `${fnFormatMeasurement}: ${measurement.value.measure}${measurement.value.unit}`,
      ),
    ).toBe(true);
    expect(result.includes(`${options.formatConfig!.selector}`)).toBe(true);
    done();
  });

  it('Get tokens - apply parsers - without custom selector', async done => {
    const options: OptionsType = {
      formatName: 'snakeCase'!,
      formatTokens: { color: 'hsl' },
    };
    const result = await toCss(seeds.tokens as Array<Token>, options, libs);
    const color = seeds.tokens.find((token: Token) => token.type === 'color') as ColorToken;
    const measurement = seeds.tokens.find(
      (token: Token) => token.type === 'measurement',
    ) as MeasurementToken;

    const fnFormatColor = libs._[options.formatName!](color.name);
    expect(
      result.includes(
        `${fnFormatColor}: ${libs
          .tinycolor(color.value as ColorValue)
          .toString(options.formatTokens?.color)}`,
      ),
    );

    const fnFormatMeasurement = libs._[options.formatName!](measurement.name);
    expect(
      result.includes(
        `${fnFormatMeasurement}: ${measurement.value.measure}${measurement.value.unit}`,
      ),
    ).toBe(true);
    expect(result.includes(`:root {`)).toBe(true);
    done();
  });

  it('Get tokens - apply parsers - with camelCase', async done => {
    const options: OptionsType = {
      formatName: 'camelCase'!,
      formatTokens: { color: 'hsl' },
    };
    const result = await toCss(seeds.tokens as Array<Token>, options, libs);
    const color = seeds.tokens.find((token: Token) => token.type === 'color') as ColorToken;
    const measurement = seeds.tokens.find(
      (token: Token) => token.type === 'measurement',
    ) as MeasurementToken;

    const fnFormatColor = libs._[options.formatName!](color.name);
    expect(
      result.includes(
        `${fnFormatColor}: ${libs
          .tinycolor(color.value as ColorValue)
          .toString(options.formatTokens?.color)}`,
      ),
    ).toBe(true);

    // Specific edge case where we don't have a space in the name but still want to parse
    expect(result.includes('colorwithaslashInit')).toBe(true);
    expect(result.includes('colorwithadashInit')).toBe(true);

    const fnFormatName = libs._.camelCase;
    expect(
      result.includes(
        `${fnFormatName(measurement.name)}: ${measurement.value.measure}${measurement.value.unit}`,
      ),
    ).toBe(true);
    expect(result.includes(`:root {`)).toBe(true);
    done();
  });
});
