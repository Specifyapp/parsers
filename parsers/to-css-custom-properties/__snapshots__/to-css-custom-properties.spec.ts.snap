// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To css Get tokens - apply parsers - all tokens 1`] = `
":root {
  /* BITMAP */
  --acme-logo: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/863c6a9c-090a-4360-bbf2-1e2026577306";
  --photo-example: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/d4ea441b-e83d-40a0-8f2d-604c28ae4c17";
  --photo-example: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/ff09d8fc-73d8-446d-91c4-772e88e776c5";
  --photo-example: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/c27d7c62-1414-49c4-ba3e-dc03aa4bd085";

  /* VECTOR */
  --activity: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/b63a7f88-0ea2-4e17-a5e8-11935562621f";
  --airplay: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/71edeeb4-6622-490e-b59a-dd768d3414a8";
  --alert-circle: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/587bd701-750c-4aa2-9b5b-843235eb5d5b";
  --alert-octagon: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/e15614aa-5e39-43c7-9a88-9ea102df8854";
  --alert-triangle: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/e4d03b25-dc16-4e25-b2b3-5ffe6a68d41d";
  --alert-triangle: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/00e59368-656f-46f1-b1bf-1200ec945dd1";
  --align-center: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/9e21c78b-dd71-42c5-9e2f-8a81f729ebd5";
  --align-center: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/0256731e-92b1-474d-b0ed-7a7d5e17338e";
  --user-mask: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/a131429d-f992-4daf-8f37-5b389b98e410";
  --user-mask: "https://figma-alpha-api.s3.us-west-2.amazonaws.com/images/4e70632a-014a-4594-84bb-6f6e18e9b2e3";

  /* DEPTH */
  --background: 1;
  --foreground: 100;
  --middle: 50;

  /* DURATION */
  --base: 300ms;
  --long: 700ms;
  --short: 100ms;
  --very-long: 3s;

  /* MEASUREMENT */
  --base-space-01: 4px;
  --base-space-02: 8px;
  --base-space-03: 12px;
  --base-space-04: 16px;
  --base-space-05: 32px;
  --base-space-06: 48px;
  --base-space-07: 64px;
  --base-space-08: 80px;
  --base-space-09: 96px;
  --base-space-10: 112px;

  /* TEXTSTYLE */
  --body-color: rgb(30, 33, 43);
  --body-font-family: Inter;
  --body-font-weight: 500;
  --body-font-size: 14px;
  --body-letter-spacing: 10px;
  --body-line-height: 20px;
  --body-text-align: left;
  --body-with-opacity-color: rgba(30, 33, 43, 0.3);
  --body-with-opacity-font-family: Inter;
  --body-with-opacity-font-weight: 500;
  --body-with-opacity-font-size: 14px;
  --body-with-opacity-letter-spacing: 10px;
  --body-with-opacity-line-height: 20px;
  --body-with-opacity-text-align: left;
  --code-color: rgb(255, 142, 5);
  --code-font-family: Fira Code;
  --code-font-weight: 500;
  --code-font-size: 13px;
  --code-line-height: 20px;
  --code-text-align: left;
  --list-font-family: Roboto;
  --list-font-weight: 400;
  --list-font-size: 14px;
  --list-font-variant: small-caps;
  --list-line-height: 20px;
  --list-text-align: left;
  --title-color: rgb(30, 33, 43);
  --title-font-family: Inter;
  --title-font-weight: 600;
  --title-font-size: 32px;
  --title-line-height: 40px;
  --title-text-align: left;
  --title-text-decoration: underline;
  --title-text-ident: 5px;
  --title-text-transform: uppercase;

  /* BORDER */
  --border-accent: 2px solid rgba(102, 80, 239, 1);
  --border-accent-with-opacity: 2px solid rgba(102, 80, 239, 0.5);
  --border-accent-without-radii: 2px solid rgba(102, 80, 239, 0.5);
  --border-dashed: 1px dashed rgba(30, 33, 43, 1);

  /* COLOR */
  --colors-accent: rgb(120, 133, 183);
  --colors-almost-black: rgb(28, 28, 28);
  --colors-black: rgb(30, 33, 43);
  --colors-green: rgb(88, 205, 82);
  --colors-grey: rgb(204, 213, 225);
  --colors-orange: rgb(255, 142, 5);
  --colors-pure-black: rgb(0, 0, 0);
  --colors-red: rgb(245, 72, 63);
  --colors-white: rgb(255, 255, 255);

  /* SHADOW */
  --elevation-1: 0px 4px 8px rgba(0, 0, 0, 0.1);
  --elevation-2: 0px 4px 28px rgba(0, 0, 0, 0.1), 0px 4px 4px rgba(0, 0, 0, 0.1);
  --elevation-3: 0px 1.8518518209457397px 3.1481480598449707px
      rgba(0, 0, 0, 0.02),
    0px 8.148148536682129px 6.518518447875977px rgba(0, 0, 0, 0.03),
    0px 20px 13px rgba(0, 0, 0, 0.04),
    0px 38.51852035522461px 25.481481552124023px rgba(0, 0, 0, 0.04),
    0px 64.81481170654297px 46.85185241699219px rgba(0, 0, 0, 0.05),
    0px 100px 80px 1px rgba(0, 0, 0, 0.07);

  /* GRADIENT */
  --gradients-colored: linear-gradient(
    90deg,
    rgb(245, 72, 63) 0%,
    rgb(255, 142, 5) 100%
  );
  --gradients-dark: linear-gradient(
    90deg,
    rgb(30, 33, 43) 0%,
    rgb(204, 213, 225) 100%
  );
  --gradients-neutral: linear-gradient(
    90deg,
    rgb(204, 213, 225) 0%,
    rgb(255, 255, 255) 100%
  );
  --gradients-safari: linear-gradient(
    90deg,
    rgb(255, 142, 5) 0%,
    rgb(255, 255, 255) 100%
  );

  /* OPACITY */
  --subtle: 0.5;
  --transparent: 0.1;
  --visible: 0.95;
}
"
`;
